src/web/mjs/connectors/templates/MH.mjs

```diff
@@ -0,0 +1,69 @@
+import Connector from '../../engine/Connector.mjs';
+import Manga from '../../engine/Manga.mjs';
+
+export default class MH extends Connector {
+
+    constructor() {
+        super();
+        super.id = undefined;
+        super.label = undefined;
+        this.tags = [];
+        this.url = undefined;
+
+        this.path = '/booklist?page=%PAGE%';
+        this.pathMatch = /page=(\d+)/;
+        this.queryMangasPageCount = 'div.page-pagination div.pagination li:nth-last-child(2) a';
+        this.queryMangas = 'ul.mh-list li div.mh-item-detali h2.title a';
+        this.queryMangaTitle = 'section.banner_detail div.info h1';
+        this.queryChapter = 'div#chapterlistload ul#detail-list-select li a';
+        this.queryPages = 'div.comiclist div.comicpage source';
+    }
+
+    async _getMangaFromURI(uri) {
+        let request = new Request(uri, this.requestOptions);
+        let data = await this.fetchDOM(request, this.queryMangaTitle);
+        let id = uri.pathname + uri.search;
+        let title = data[0].textContent.trim();
+        return new Manga(this, id, title);
+    }
+
+    async _getMangas() {
+        let mangaList = [];
+        let request = new Request(new URL(this.path.replace('%PAGE%', 1), this.url), this.requestOptions);
+        let data = await this.fetchDOM(request, this.queryMangasPageCount);
+        let pageCount = parseInt(data[0].href.match(this.pathMatch)[1]);
+        for(let page = 1; page <= pageCount; page++) {
+            let mangas = await this._getMangasFromPage(page);
+            mangaList.push(...mangas);
+        }
+        return mangaList;
+    }
+
+    async _getMangasFromPage(page) {
+        let request = new Request(new URL(this.path.replace('%PAGE%', page), this.url), this.requestOptions);
+        let data = await this.fetchDOM(request, this.queryMangas);
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
+                title: element.text.trim()
+            };
+        });
+    }
+
+    async _getChapters(manga) {
+        let request = new Request(new URL(manga.id, this.url), this.requestOptions);
+        let data = await this.fetchDOM(request, this.queryChapter);
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
+                title: element.childNodes[0].nodeValue.trim()
+            };
+        });
+    }
+
+    async _getPages(chapter) {
+        let request = new Request(new URL(chapter.id, this.url), this.requestOptions);
+        let data = await this.fetchDOM(request, this.queryPages);
+        return data.map(element => this.getAbsolutePath(element.dataset.echo || element, request.url));
```

Comment:
May extend options to cover cases from `iqiyi ` and `mkzhan`, so these websites don't need to overwrite it
`this.getAbsolutePath(element.dataset.src || element.dataset.original || element.dataset.echo || element, request.url)`